libr/debug/p/native/linux/linux_coredump.c

```diff
@@ -0,0 +1,1434 @@
+/* radare - LGPL - Copyright 2009-2016 - pancake */
+
+#include <r_debug.h>
+#include <sys/ptrace.h>
+#include <sys/uio.h>
+#include "linux_coredump.h"
+
+static map_file_t mapping_file = {0,0};
+
+static inline char *prpsinfo_get_fname(char *buffer) {
+	/* buffer contains: str\0str1\0str2\0... Here we're only interested in the first part of string, and strdup copies till it reaches a \0 */
+	return strdup (buffer);
+}
+
+static char *get_basename(const char *pfname, int len) {
+	char *p;
+
+	for (p = pfname + len; p != pfname ; p--) {
+		if (*p == '/') return (p + 1);
+	}
+
+	return p;
+}
+
+static char *prpsinfo_get_psargs(char *buffer, char *pfname, int size_psargs, int len) {
+	char paux[ELF_PRARGSZ];
+	char *p;
+	int i;
+	int bytes_left;
+
+	p = r_mem_dup (pfname, size_psargs);
+	if (!p) return NULL;
+
+	bytes_left = strlen (pfname);
+	buffer = strchr (buffer, '\0');
+
+	for(i = 0; i + bytes_left < len && i + bytes_left < (size_psargs - 1); i++) {
```

Comment:
Why the check is on `i + bytes_left` but `paux` is accessed through `i`? Also, I would put a check on the maximum size of paux (ELF_PRARGSZ)